src/components/UserManagement.tsx

```diff
@@ -6,23 +6,192 @@
+                    props.level === 'User' ? 'selected' : 'deselected'
+                }`}
+            >
+                user
+            </span>
+        </div>
+    );
+}
+
+interface accountControlsProps {
+    uid: string;
+    email: string;
+}
+
+function AccountControls(props: accountControlsProps) {
+    return (
+        <div className={`accountControls`}>
+            <span
+                className="delete"
+                onClick={() => {
+                    if (window.confirm(`Delete user: ${props.email}`)) {
+                        deleteUser(props.uid);
+                    }
+                }}
+            >
+                delete
+            </span>
+        </div>
+    );
+}
+
 export default function UserManagement() {
     const [authorized, setAuthorization] = useState(false);
+    const users = useSelector((state: RootState) => state.users);
 
     getUserPrivilege().then((privilege) => {
         if (privilege === 'Owner' || privilege === 'Admin') {
             setAuthorization(true);
         }
     });
 
+    function promptForEmail() {
+        let result = prompt(
+            "Enter the email address for the account you'd like registered",
+            'email@somewhere'
+        );
+        registerUser(result as string);
+    }
+
+    const columns = [
+        {
+            name: 'email',
+            header: <b>email address</b>,
+            defaultFlex: 2,
+            editable: false,
+        },
+        { name: 'phone', header: 'phone number', defaultFlex: 2 },
+        {
+            name: 'privileges',
+            header: 'privileges',
+            editable: false,
+            defaultFlex: 2,
+        },
+        {
+            name: 'controls',
+            header: 'caution',
+            defaultFlex: 1,
+            editable: false,
+        },
+    ];
+
+    let data: any = [];
+
+    for (const [uid, user] of Object.entries(users)) {
+        let userData = user as User;
+        let phoneNumber = userData.phoneNumber;
+        if (phoneNumber == null) {
+            phoneNumber = '';
+        }
+        if (userData.userGroup != 'Owner' && !userData.disabled) {
+            data.push({
+                email: userData.email,
+                phone: phoneNumber,
+                privileges: (
+                    <PrivilegeToggle level={userData.userGroup} uid={uid} />
+                ),
+                uniqueId: uid,
+                controls: <AccountControls uid={uid} email={userData.email} />,
+            });
+        }
+    }
+
+    const onEditComplete = (info: TypeEditInfo) => {
+        switch (info.columnId) {
+            case 'phone': {
+                editPhoneNumber(info.rowId, info.value);
+                break;
+            }
+            case 'email': {
+                editEmail(info.rowId, info.value);
+                break;
+            }
+        }
+    };
+
+    const gridStyle = { minHeight: 550 };
+
     return (
         <div className="userManagement w-full">
```

Comment:
The table runs off the bottom of the screen but I'm not sure where the styling is for that.

Reply:
I'm not sure why it's doing this. It didn't do this when I made the pull request. I'll check it out